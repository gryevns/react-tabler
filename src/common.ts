export type Color =
    | "blue"
    | "azure"
    | "indigo"
    | "purple"
    | "pink"
    | "red"
    | "orange"
    | "yellow"
    | "lime"
    | "green"
    | "teal"
    | "cyan"
    | "gray";

export const colors = [
    "blue" as const,
    "azure" as const,
    "indigo" as const,
    "purple" as const,
    "pink" as const,
    "red" as const,
    "orange" as const,
    "yellow" as const,
    "lime" as const,
    "green" as const,
    "teal" as const,
    "cyan" as const,
    "gray" as const,
];

export type Variant =
    | "primary"
    | "secondary"
    | "success"
    | "info"
    | "warning"
    | "danger";

export const variants = [
    "primary" as const,
    "secondary" as const,
    "success" as const,
    "info" as const,
    "warning" as const,
    "danger" as const,
];

export type Size = "sm" | "md" | "lg" | "xl";

export const sizes = ["sm" as const, "md" as const, "lg" as const, "xl" as const];
