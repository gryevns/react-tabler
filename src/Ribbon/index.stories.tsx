import React from "react";
import { Star } from "react-feather";

import Card from "../Card";
import Grid from "../Grid";
import Ribbon from "../Ribbon";
import { colors } from "../common";

export default {
    title: "Ribbon",
};

export const Position = () => {
    const positions = [
        ["top"],
        ["right"],
        ["bottom"],
        ["bottom", "left"],
        ["left"],
        ["top", "left"],
    ];
    return (
        <Grid.Row>
            {positions.map(position => {
                const props = position.reduce((result, p) => {
                    result[p] = true;
                    return result;
                }, {});
                return (
                    <Grid.Col width={4}>
                        <Card>
                            <Card.Body className="h-8"></Card.Body>
                            <Ribbon {...props}>
                                <Star size={18} color="white" />
                            </Ribbon>
                        </Card>
                    </Grid.Col>
                );
            })}
        </Grid.Row>
    );
};

export const Colors = () => (
    <Grid.Row>
        {colors.map(color => (
            <Grid.Col>
                <Ribbon color={color}>
                    <Star size={18} color="white" />
                </Ribbon>
            </Grid.Col>
        ))}
    </Grid.Row>
);

export const Bookmark = () => (
    <Ribbon bookmark>
        <Star size={18} color="white" />
    </Ribbon>
);
