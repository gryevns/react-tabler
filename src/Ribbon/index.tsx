import * as React from "react";
import cn from "classnames";
import { Color } from "../common";

export interface RibbonProps {
    bookmark?: boolean;
    children?: React.ReactNode;
    color?: Color;
    top?: boolean;
    right?: boolean;
    bottom?: boolean;
    left?: boolean;
}

export const Ribbon = ({
    children,
    top,
    right,
    bottom,
    left,
    bookmark,
    color,
}: RibbonProps) => {
    const classes = cn("ribbon", {
        "ribbon-top": top,
        "ribbon-right": right,
        "ribbon-bottom": bottom,
        "ribbon-left": left,
        [`bg-${color}`]: !!color,
        "ribbon-bookmark": bookmark,
    });
    return <div className={classes}>{children}</div>;
};

export default Ribbon;
