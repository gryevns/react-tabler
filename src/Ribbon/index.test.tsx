import React from "react";
import { shallow } from "enzyme";

import { colors, Color } from "../common";
import Ribbon from "./index";

describe("Ribbon", () => {
    it("should match color snapshot", () => {
        colors.map((color: Color) => {
            const component = shallow(<Ribbon color={color} />);
            expect(component).toMatchSnapshot();
        });
    });
    it("should match top snapshot", () => {
        const component = shallow(<Ribbon top />);
        expect(component).toMatchSnapshot();
    });
    it("should match right snapshot", () => {
        const component = shallow(<Ribbon right />);
        expect(component).toMatchSnapshot();
    });
    it("should match bottom snapshot", () => {
        const component = shallow(<Ribbon bottom />);
        expect(component).toMatchSnapshot();
    });
    it("should match left snapshot", () => {
        const component = shallow(<Ribbon left />);
        expect(component).toMatchSnapshot();
    });
    it("should match bookmark snapshot", () => {
        const component = shallow(<Ribbon bookmark />);
        expect(component).toMatchSnapshot();
    });
});
