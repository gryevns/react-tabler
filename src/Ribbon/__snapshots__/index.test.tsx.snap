// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon should match bookmark snapshot 1`] = `
<div
  className="ribbon ribbon-bookmark"
/>
`;

exports[`Ribbon should match bottom snapshot 1`] = `
<div
  className="ribbon ribbon-bottom"
/>
`;

exports[`Ribbon should match color snapshot 1`] = `
<div
  className="ribbon bg-blue"
/>
`;

exports[`Ribbon should match color snapshot 2`] = `
<div
  className="ribbon bg-azure"
/>
`;

exports[`Ribbon should match color snapshot 3`] = `
<div
  className="ribbon bg-indigo"
/>
`;

exports[`Ribbon should match color snapshot 4`] = `
<div
  className="ribbon bg-purple"
/>
`;

exports[`Ribbon should match color snapshot 5`] = `
<div
  className="ribbon bg-pink"
/>
`;

exports[`Ribbon should match color snapshot 6`] = `
<div
  className="ribbon bg-red"
/>
`;

exports[`Ribbon should match color snapshot 7`] = `
<div
  className="ribbon bg-orange"
/>
`;

exports[`Ribbon should match color snapshot 8`] = `
<div
  className="ribbon bg-yellow"
/>
`;

exports[`Ribbon should match color snapshot 9`] = `
<div
  className="ribbon bg-lime"
/>
`;

exports[`Ribbon should match color snapshot 10`] = `
<div
  className="ribbon bg-green"
/>
`;

exports[`Ribbon should match color snapshot 11`] = `
<div
  className="ribbon bg-teal"
/>
`;

exports[`Ribbon should match color snapshot 12`] = `
<div
  className="ribbon bg-cyan"
/>
`;

exports[`Ribbon should match color snapshot 13`] = `
<div
  className="ribbon bg-gray"
/>
`;

exports[`Ribbon should match left snapshot 1`] = `
<div
  className="ribbon ribbon-left"
/>
`;

exports[`Ribbon should match right snapshot 1`] = `
<div
  className="ribbon ribbon-right"
/>
`;

exports[`Ribbon should match top snapshot 1`] = `
<div
  className="ribbon ribbon-top"
/>
`;
